src/styles/index.scss

```diff
@@ -1,3 +1,86 @@
+$days: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+
+@each $day, $marginMultiplier in $days {
+  .square-#{$day} {
+    $baseSize: 100px;
+    $gap: 1px;
+    $size: $baseSize * $marginMultiplier + $gap;
+
+    width: $size;
+    height: $size;
+    margin: $gap;
+  }
+}
+
 body {
   margin: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+.calendar {
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 706px;
+  display: flex;
+}
+
+.calendar__day {
+  box-sizing: border-box;
+  text-align: center;
+  background-color: #eee;
+  border: 1px solid #000;
+  align-content: center;
+  width: 100px;
+  height: 100px;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  transition: all 0.5s;
+}
+
+@each $day, $index in $days {
+  .calendar--start-day-#{$day} .calendar__day:nth-child(1) {
+    margin-left: calc(100px * $index + 1px * $index);
+  }
+}
+
+.calendar__day::before {
+  content: number;
```

Comment:
The `content` property in CSS should be a string or a value that can be evaluated to a string. The value 'number' is not valid here. Consider using a valid string or a dynamic value.